### ISSUE TITLE
Nova 3.28 index refresh breaks Action::modal

### ISSUE THREAD

**Issue by Tarpsvo** [laravel/nova-issues#3511]
- Laravel Version: 8.12.0
- Nova Version: 3.28.0
- PHP Version: 7.4.0
- Database Driver & Version: MySQL 5.7

### Description:
This new "feature" in the 3.28 changelog:
```
Nova will now refresh the parent index view when executing actions with openInNewTab, download and modal methods.
```

breaks `Action::modal()` as the `<action-selector>` component that renders the modals gets unmounted when the whole table refreshes on the index view.

### Detailed steps to reproduce the issue on a fresh Nova installation:

Create a non-standalone action that returns an Action::modal() and watch your modal never get rendered.

**Comment by Tarpsvo** [comment]
I suggest to move the code that renders the response action modal to a place where it can't be unmounted so easily. This will also make it work with actions that are rendered on the table row (where no action-selector is rendered).

```
<component
        :is="actionResponseData.modal"
        @close="closeActionResponseModal"
        v-if="showActionResponseModal"
        :data="actionResponseData"
      />
```

**Comment by Tarpsvo** [comment]
Hi! What was the resolution? :) Thanks.

**Comment by crynobone** [comment]
PR has been merged and pending next release

**Comment by crynobone** [comment]
This has been fixed with v3.29.0. https://nova.laravel.com/releases/3.29.0
